### PR TITLE
Update Yoast PHPUnit Polyfill to 2.0

### PR DESCRIPTION
**Pull request by jonathansadowski**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

PHPUnit has been failing on nightly. Updating to a more-recent version of the PHPUnit Polyfill seems to address the issue.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Verify PHPUnit passes in the checks below, especially against nightly. For it failing prior to this change, see [here](https://github.com/woocommerce/woocommerce/actions/runs/5929531374/job/16078237000?pr=39828).
2.
3.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 
This PR updates a dev dependency and a changelog is not required.

</details>

### CODE REVIEW

**Comment by rrennick** [comment]
@jonathansadowski composer.lock has a merge conflict

**Comment by jonathansadowski** [comment]
Thanks @rrennick, just resolved those.

**Review by rrennick — APPROVED** [review]
@jonathansadowski This looks good. Approving but let's wait until CI passes before merging.

**Pull request merged**
